cityofaustin/joplin
# 4097 permissions search view

# Description

Adds a method based on explorable_pages in wagtail that will return page a user can explore based on their department(s)

Explorable pages is very inclusive and includes all descendants of explorable pages in the queryset that is returned. Editors and Moderators have permissions on the home page, thus the original method returns all pages. 

This PR only limits what is viewable in the search/index view, it does not address preventing users from editing pages if they have the link to a page. I stumbled across that solution, its commented out here just to isolate the behavior of the dept_explorable_pages method.









# Types of changes



# How can this be tested?



 

https://joplin-pr-4097-perm-view.herokuapp.com/admin/pages/14/edit/#tab-content

kitchen@austintexas.io pw: k <-- only see kitchen pages
health@austintexas.io pw: p <-- only see aph pages

# Checklist:
